### PR TITLE
Fix pytest 5 support

### PR DESCRIPTION
**Pull request by ollipa**
Fixes Pytest 5 support. I also modified nox tests so we are now testing multiple pytest versions.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -14,7 +14,9 @@ python =
 
 [testenv]
 deps =
-    pytest
+    pytest5: pytest>=5.0,<5.9
+    pytest6: pytest>=6.0,<6.9
```
Comment (christophe-riolo) [inline tox.ini]: why limit to 5.9 or 6.9?

Comment (ollipa) [reply tox.ini]: Right, it should be 6.0 and 7.0.

Comment (ollipa) [reply tox.ini]: Fixed: https://github.com/flexmock/flexmock/pull/51/commits/2092441f9ed60151666e03db0c9008a6cde2bfaf

**Review by christophe-riolo — APPROVED** [review]
Looking good 👍

**Pull request merged**